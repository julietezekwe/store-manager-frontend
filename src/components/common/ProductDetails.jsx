import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getAproduct } from '../../actions/productsAction';

export class ProductDetail extends Component {
  async componentDidMount() {
    await this.props.getAproduct(Number(this.props.match.params.id));
  }

  render() {
    const { productDetail } = this.props;
    return (
      <div className=" description">
        <div className="col-5 " style={{ marginTop: '30px' }}>
          <img src={productDetail && productDetail.image} alt="Products" />
        </div>
        <br />
        <h2>{productDetail && productDetail.productname}</h2>
        <h3>
          {' '}
          Description: <p>{productDetail && productDetail.description}</p>{' '}
        </h3>
        <br />
        <h3>
          Stock : <p> {productDetail && productDetail.quantity} </p>
        </h3>{' '}
        <br />
        <h3>
          Price : <p>{productDetail && productDetail.price}</p>{' '}
        </h3>{' '}
        <br />
        <h3>
          minimum inventory quantity : <p>{productDetail && productDetail.min}</p>{' '}
        </h3>{' '}
        <br />
        <h3>
          Creation Date : <p>{productDetail && productDetail.created_at}</p>{' '}
        </h3>{' '}
        <br />
        <h3>
          Category : <p>{productDetail && productDetail.category}</p>{' '}
        </h3>{' '}
        <br />
        <h3>Features</h3>
        <p>Long Lasting, </p>
        <p>Durable, </p>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  productDetail: state.products.productDetail[0],
});
export default connect(mapStateToProps, { getAproduct })(ProductDetail);
