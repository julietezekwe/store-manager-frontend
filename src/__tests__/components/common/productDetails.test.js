/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import { ProductDetail } from '../../../components/common/ProductDetails.jsx';

let wrapper;
const props = {
  getAproduct: () => Promise.resolve(),
  productDetail: {
    productid: 1,
    productname: 'shoes',
    description: 'fine',
    quantity: 10,
    price: 1200,
    min: 1,
    image: 'imag.url',
  },
};
beforeEach(() => {
  wrapper = mount(<ProductDetail {...props} />);
});
describe('ProductDetail Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
