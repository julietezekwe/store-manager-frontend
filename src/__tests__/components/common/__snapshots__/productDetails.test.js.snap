// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetail Component should match snapshot 1`] = `
<ProductDetail
  getAproduct={[Function]}
  productDetail={
    Object {
      "description": "fine",
      "image": "imag.url",
      "min": 1,
      "price": 1200,
      "productid": 1,
      "productname": "shoes",
      "quantity": 10,
    }
  }
>
  <div
    className=" description"
  >
    <div
      className="col-2 "
      style={
        Object {
          "marginLeft": "80px",
          "marginRight": "40px",
          "marginTop": "30px",
        }
      }
    >
      <img
        alt="Products"
        src="imag.url"
      />
    </div>
    <br />
    <h2>
      shoes
    </h2>
    <h3>
       
      Description: 
      <p>
        fine
      </p>
       
    </h3>
    <br />
    <h3>
      Stock : 
      <p>
         
        10
         
      </p>
    </h3>
     
    <br />
    <h3>
      Price : 
      <p>
        1200
      </p>
       
    </h3>
     
    <br />
    <h3>
      minimum inventory quantity : 
      <p>
        1
      </p>
       
    </h3>
     
    <br />
    <h3>
      Creation Date : 
      <p />
       
    </h3>
     
    <br />
    <h3>
      Category : 
      <p />
       
    </h3>
     
    <br />
    <h3>
      Features
    </h3>
    <p>
      Long Lasting, 
    </p>
    <p>
      Durable, 
    </p>
  </div>
</ProductDetail>
`;
